src/Polly.Contrib.Simmy/AsyncMonkeyEngine.cs

```diff
@@ -41,10 +41,15 @@ internal static class AsyncMonkeyEngine
             Func<Context, CancellationToken, Task> injectedBehaviour,
             Func<Context, CancellationToken, Task<Double>> injectionRate,
             Func<Context, CancellationToken, Task<bool>> enabled,
+            Func<Context, CancellationToken, Task> beforeInjectCallback,
             bool continueOnCapturedContext)
         {
             if (await ShouldInjectAsync(context, cancellationToken, injectionRate, enabled, continueOnCapturedContext).ConfigureAwait(continueOnCapturedContext))
             {
+                if (beforeInjectCallback != null)
+                {
+                    await beforeInjectCallback(context, cancellationToken);
```

Comment:
now that we're allowing injecting a delegate right before injecting the chaos, I think we should honor the token, just in case the user signaled the token from there:
```
cancellationToken.ThrowIfCancellationRequested();
```